### PR TITLE
Update to containerd 2.0 rc

### PR DESCRIPTION
**Pull request by dmcgowan**
Use a containerd 2.0 tag to use v2 packages. To find broken areas before final release.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -12,7 +12,7 @@
 #   See the License for the specific language governing permissions and
 #   limitations under the License.
 
-ARG CONTAINERD_VERSION=v1.7.13
+ARG CONTAINERD_VERSION=v2.0.0-rc.0
```
Comment (ktock) [inline Dockerfile:15]: Could you bump up this to v2.0.0-rc.1?

File: Dockerfile [hunk 2]
```diff
@@ -33,7 +33,7 @@ ARG CRI_TOOLS_VERSION=v1.30.0
 # Legacy builder that doesn't support TARGETARCH should set this explicitly using --build-arg.
 # If TARGETARCH isn't supported by the builder, the default value is "amd64".
 
-FROM golang:1.21.5-bullseye AS golang-base
+FROM golang:1.22.3-bullseye AS golang-base
```
Comment (ktock) [inline Dockerfile:36]: On ` runc-dev` stage (line 81), please use go 1.21 to fix the current CI failure (`OCI runtime create failed: runc create failed: unable to start container process: error during container init: read init-p: connection reset by peer: unknown`).

**Comment by ktock** [comment]
I've sent patches to solve the CI failures:

- https://github.com/containerd/containerd/pull/10287 is for fixing https://github.com/containerd/stargz-snapshotter/actions/runs/9327851434/job/25678303530?pr=1545#step:4:10542
-  #1695 is for fixing https://github.com/containerd/stargz-snapshotter/actions/runs/9327851434/job/25678303459?pr=1545

**Comment by ktock** [comment]
@dmcgowan Another PR for containerd v2 migration (#1722) is posted recently.
If this PR is still active, please let me know and rebase this PR so that we can prioritize this PR.

**Comment by AkihiroSuda** [comment]
#1722 is now merged

**Pull request closed without merging**